### PR TITLE
Shadow DOM: Use the new getRootNode API

### PR DESCRIPTION
**Pull request by dvoytenko**

### CODE REVIEW

File: src/shadow-embed.js [hunk 1]
```diff
@@ -17,10 +17,14 @@
 import {ShadowCSS} from '../third_party/webcomponentsjs/ShadowCSS';
 import {ampdocServiceFor} from './ampdoc';
 import {dev} from './log';
-import {escapeCssSelectorIdent} from './dom';
+import {closestNode, escapeCssSelectorIdent} from './dom';
 import {extensionsFor} from './extensions';
 import {insertStyleElement} from './style-installer';
 
+/**
+ * Used for non-composed root-node search. See `getRootNode`.
```
Comment (aghassemi) [inline src/shadow-embed.js]: @cosnt {!GetRootNodeOptions}

File: src/shadow-embed.js [hunk 2]
```diff
@@ -111,6 +115,39 @@ function createShadowRootPolyfill(hostElement) {
...
+  if (!value) {
+    return false;
+  }
+  // Node.nodeType == DOCUMENT_FRAGMENT to speed up the tests. Unfortunately,
+  // nodeType of DOCUMENT_FRAGMENT is used currently for ShadowRoot nodes.
+  if (value.tagName == 'I-AMP-SHADOW-ROOT') {
+    return true;
+  }
+  return (value.nodeType == /* DOCUMENT_FRAGMENT */ 11 &&
+      Object.prototype.toString.call(value) === '[object ShadowRoot]');
```
Comment (aghassemi) [inline src/shadow-embed.js]: yikes :) I hope adding a nodetype for this is being considered.

File: src/shadow-embed.js [hunk 3]
```diff
@@ -111,6 +115,39 @@ function createShadowRootPolyfill(hostElement) {
...
+}
+
+
+/**
+ * Return shadow root for the specified node.
+ * @param {!Node} node
+ * @return {?ShadowRoot}
+ */
+export function getShadowRootNode(node) {
+  if (isShadowDomSupported() && Node.prototype.getRootNode) {
```
Comment (aghassemi) [inline src/shadow-embed.js]: `isShadowDomSupported()` seems somewhat redundant, I assume if browser supports getRootNode, they satisfy the condition for `isShadowDomSupported()` anyway.

File: test/functional/test-shadow-embed.js [hunk 4]
```diff
@@ -153,6 +152,68 @@ describe('shadow-embed', () => {
...
+    it('should yield true for natively-supported createShadowRoot API', () => {
+      const element = document.createElement('div');
+      if (element.createShadowRoot) {
+        const shadowRoot = element.createShadowRoot();
+        expect(isShadowRoot(shadowRoot)).to.be.true;
+      }
+    });
+
+    it('should yield false for non-document-fragment shadow root', () => {
+      const element = document.createElement('div');
```
Comment (aghassemi) [inline test/functional/test-shadow-embed.js]: this seems the same as the test above for "other types of nodes"

File: test/functional/test-shadow-embed.js [hunk 5]
```diff
@@ -153,6 +152,68 @@ describe('shadow-embed', () => {
...
+      expect(isShadowRoot(element)).to.be.false;
+    });
+
+    it('should yield false for document-fragment shadow root', () => {
```
Comment (aghassemi) [inline test/functional/test-shadow-embed.js]: s/shadow root/node ?

File: src/shadow-embed.js (hunk 3 above)
Comment (dvoytenko) [inline src/shadow-embed.js]: Nope :) I could tell some stories here...

File: src/shadow-embed.js (hunk 2 above)
Comment (dvoytenko) [inline src/shadow-embed.js]: Not that I heard :( Will add to my whatwg list.

File: test/functional/test-shadow-embed.js (hunk 4 above)
Comment (dvoytenko) [inline test/functional/test-shadow-embed.js]: Correct. Removed.

File: test/functional/test-shadow-embed.js (hunk 5 above)
Comment (dvoytenko) [inline test/functional/test-shadow-embed.js]: Done

File: src/shadow-embed.js (hunk 1 above)
Comment (dvoytenko) [inline src/shadow-embed.js]: Done

File: src/shadow-embed.js (hunk 3 above)
Comment (aghassemi) [inline src/shadow-embed.js]: hehe, oh Web..

**Pull request merged**